server.js

```diff
@@ -1,23 +1,152 @@
-const express = require("express");
-const cors = require("cors");
+const express = require ('express');
+const cors = require ('cors');
+const Joi = require ('joi');
 
-const app = express();
+const app = express ();
+app.use (express.json ());
+
+app.use (cors ());
 
-app.use(cors());
 
 const welcomeMessage = {
   id: 0,
-  from: "Bart",
-  text: "Welcome to CYF chat system!",
+  from: 'Bart',
+  text: 'Welcome to CYF chat system!',
+
+  timeStamp: Date().toLocaleString (),
 };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+const messages = [
+  welcomeMessage,
+  {
+    id: 1,
+    from: 'Taslima',
+    text: 'Welcome to our code your future',
+    timestamp: new Date ().toLocaleString (),
+  },
+  {
+    id: 2,
+    from: 'Aysha',
+    text: 'Welcome to my nursery!!!',
+    timestamp: new Date ().toLocaleString (),
+  },
+  {
+    id: 3,
+    from: 'Khadijah',
+    text: 'Welcome to my school!!!',
+    timestamp: new Date ().toLocaleString (),
+  },
+];
+
+app.get ('/', function (request, response) {
+  response.sendFile (__dirname + '/index.html');
+});
+
+//read all messages
+app.get ('/messages', function (request, response) {
+  response.send (messages);
+});
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
+//create a new message.
+app.post ('/messages', (request, response) => {
+  //error handling using joi package manager
+  const schema = {
+    from: Joi.string ().min (3).required (),
+    text: Joi.string ().min (3).required (),
+  };
+  const result = Joi.validate (request.body, schema);
+  if (result.error) {
+    response.status (400).send (result.error.details[0].message);
+    return;
+  }
+  const message = {
+    id: messages.length + 1,
+    from: request.body.from,
+    text: request.body.text,
+    timestamp: new Date ().toLocaleString (),
+  };
+  messages.push (message);
+  response.send (message);
 });
 
-app.listen(process.env.PORT);
+//delete a message by id
+//look up the course
+//not existing,return 404
+
+app.delete ('/messages/:id', (req, res) => {
+  const messageId = req.params.id;
+  //validation
+  const message = messages.find (
+    message => message.id === parseInt (messageId)
+  );
+  if (!message)
+    res.status (404).send ('The message with given Id does not exist');
+  //delete
+  const index = messages.indexOf (message);
+  messages.splice (index, 1);
+  //return response back to user
+  res.send (message);
+});
```

Comment:
Excellent. The only thing I'd suggest is it's not necessary to send the deleted message back to the client, it'd be fine just to send an acknowledgment that the message has been deleted.